fmt/src/comments.rs

```diff
@@ -65,34 +78,46 @@ impl CommentWithMetadata {
         last_comment: Option<&CommentWithMetadata>,
     ) -> Self {
         let src_before = &src[..comment.loc().start()];
+        if src_before.is_empty() {
+            return Self::new(comment, CommentPosition::Prefix, false, 0)
+        }
+
         let mut lines_before = src_before.lines().rev();
         let this_line =
             if src_before.ends_with('\n') { "" } else { lines_before.next().unwrap_or("") };
         let indent_len = this_line.chars().take_while(|c| c.is_whitespace()).count();
+        let last_line = lines_before.next();
+
+        if matches!(comment, Comment::DocLine(..) | Comment::DocBlock(..)) {
+            return Self::new(
+                comment,
+                CommentPosition::Prefix,
+                last_line.map(|line| line.trim_start().is_empty()).unwrap_or(false),
```

Comment:
maybe `map_or` would be cleaner?

Reply:
personally, not a fan of `map_or` because it trips me up that the alternative is the first arg.

but no strong feelings about it, `unwrap_or_default` also fine here

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some_and <- this seems like it would be useless but i feel like i'll use it all the time when it gets stabilized

Reply:
yup, guess just style pref. resolving this